packages/calcite-components/src/components/list/list.e2e.ts

```diff
@@ -373,6 +377,51 @@ describe("calcite-list", () => {
     expect(visibleItems.map((item) => item.id)).toEqual(["label-match", "description-match", "value-match"]);
   });
 
+  it("filters initially with matchFields", async () => {
+    const page = await newE2EPage();
+    await page.setContent(html`
+      <calcite-list filter-enabled filter-text="match">
+        <calcite-list-item
+          id="label-match"
+          label="match"
+          description="description-1"
+          value="value-1"
+        ></calcite-list-item>
+        <calcite-list-item
+          id="description-match"
+          label="label-2"
+          description="match"
+          value="value-1"
+        ></calcite-list-item>
+        <calcite-list-item
+          id="value-match"
+          label="label-3"
+          description="description-3"
+          value="match"
+        ></calcite-list-item>
+        <calcite-list-item
+          id="no-match"
+          label="label-4"
+          description="description-4"
+          value="value-4"
+        ></calcite-list-item>
+      </calcite-list>
+    `);
+
+    await page.waitForChanges();
+    const list = await page.find("calcite-list");
+    list.setProperty("matchFields", ["label", "description"]);
+    await page.waitForChanges();
+    await page.waitForTimeout(listDebounceTimeout);
```

Comment:
Is there an event we can listen for? This is fine for now, but we should start moving away from timeouts as [`puppeteer@22.2.0` removed `page.waitForTimeout`](https://github.com/puppeteer/puppeteer/blob/7966dd726288bda48650a3604f8e12c6e4be41db/packages/puppeteer-core/CHANGELOG.md#2200-2024-02-05).

Reply:
Not currently, we would need some kind of event for when filteredItems is updated. Can you create an issue to add this so we can clean up these tests?

Reply:
🔥https://github.com/Esri/calcite-design-system/issues/9661🔥